### PR TITLE
Fix subject revision list not showing

### PR DESCRIPTION
**Pull request by mpgxvii**
- Fixes front end issue (even when backend sends correct results, the front end isn't showing the revisions)
- Simplify subject revisions endpoint

Aims to solve  #558

### CODE REVIEW

File: src/main/java/org/radarcns/management/web/rest/SubjectResource.java
Comment (mpgxvii) [inline src/main/java/org/radarcns/management/web/rest/SubjectResource.java:336]: @yatharthranjan @blootsvoets @nivemaham Would it be alright to remove this mapping and just check for permissions of the subject once since the revision results are only the specified subject's revisions (in `getRevisionsForEntity`)?

Comment (blootsvoets) [reply src/main/java/org/radarcns/management/web/rest/SubjectResource.java:336]: It could be that this is due to subjects moving from one project to another. We generally don't do this, so it might be fine to remove it...

**Review by yatharthranjan — DISMISSED** [review]
Looks fine to me, but probably Joris or Nivethika should review before merging.

File: src/main/java/org/radarcns/management/web/rest/SubjectResource.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/management/web/rest/SubjectResource.java:334]: This will throw NPE if project is null. better to throw intentionally in .orElse. I’m not sure about the implementation of checkpermissiononsubject if the project name is null. In that case, only a user or client with global level access should be able to see the data.

File: src/main/java/org/radarcns/management/web/rest/SubjectResource.java
Comment (blootsvoets) [inline src/main/java/org/radarcns/management/web/rest/SubjectResource.java:331]: You could add a .map(p -> p.getProjectName()) here.

**Review by blootsvoets — APPROVED** [review]
LGTM

**Pull request merged**
